sources/packages/web/src/views/student/application-offering-change-request/ApplicationOfferingChangeFormView.vue

```diff
@@ -0,0 +1,201 @@
+    <template #tab-header>
+      <v-tabs stacked color="primary"
+        ><v-tab
+          v-for="item in items"
+          :key="item.label"
+          :to="item.command()"
+          :ripple="false"
+        >
+          <span class="mx-2 label-bold"> {{ item.label }} </span>
+        </v-tab>
+      </v-tabs>
+    </template>
+    <router-view />
+    <approve-application-offering-change-request-modal
+      ref="approveApplicationOfferingChangeRequestModal"
+    />
+    <decline-application-offering-change-request-modal
+      ref="declineApplicationOfferingChangeRequestModal"
+    />
+  </full-page-container>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed, onMounted, watchEffect } from "vue";
+import { RouteLocationRaw } from "vue-router";
+import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
+import ApproveApplicationOfferingChangeRequestModal from "@/components/aest/students/modals/ApproveApplicationOfferingChangeRequestModal.vue";
+import DeclineApplicationOfferingChangeRequestModal from "@/components/aest/students/modals/DeclineApplicationOfferingChangeRequestModal.vue";
+import { ModalDialog, useSnackBar } from "@/composables";
+import { ApplicationOfferingChangeRequestService } from "@/services/ApplicationOfferingChangeRequestService";
+import { UpdateApplicationOfferingChangeRequestAPIInDTO } from "@/services/http/dto";
+import { ApplicationOfferingChangeRequestStatus } from "@/types";
+import router from "@/router";
+
+export default defineComponent({
+  components: {
+    ApproveApplicationOfferingChangeRequestModal,
+    DeclineApplicationOfferingChangeRequestModal,
+  },
+  props: {
+    applicationOfferingChangeRequestId: {
+      type: Number,
+      required: true,
+    },
+    applicationId: {
+      type: Number,
+      required: false,
+    },
+  },
+  setup(props) {
+    const snackBar = useSnackBar();
+    const applicationOfferingChangeRequestStatus =
+      ref<ApplicationOfferingChangeRequestStatus>();
+    const items = ref([
+      {
+        label: "Requested change",
+        command: () => ({
+          name: StudentRoutesConst.STUDENT_REQUESTED_APPLICATION_OFFERING_CHANGE,
+          params: {
+            applicationOfferingChangeRequestId:
+              props.applicationOfferingChangeRequestId,
+            applicationId: props.applicationId,
+          },
+        }),
+      },
+      {
+        label:
+          applicationOfferingChangeRequestStatus.value ===
+          ApplicationOfferingChangeRequestStatus.Approved
+            ? "Previous application details"
+            : "Active application details",
+        command: () => ({
+          name: StudentRoutesConst.STUDENT_ACTIVE_APPLICATION_DETAILS,
+          params: {
+            applicationOfferingChangeRequestId:
+              props.applicationOfferingChangeRequestId,
+            applicationId: props.applicationId,
+          },
+        }),
+      },
+    ]);
+    watchEffect(async () => {
```

Comment:
why didn't you use `watch` with `immediate`? What is the benefit of `watchEffect` that we can't achieve by using `watch`?

Reply:
The benefit of `watchEffect` is that we don't have to manually declare the reactive values that we want to be watched and vue tracks them automatically for us. This thus prevents the need to explicitly specify the list of dependencies that we want to track. In addition to this, `watchEffect` will execute immediately, thus removing the need to use `immediate: true` for the purpose of eager loading.

IMO, we should use `watch` when we want to track only a few reactive values of all the reactive values provided in the callback. Otherwise, if there are a lot of reactive dependencies we are tracking in the callback, we would be specifying all of them manually.

@dheepak-aot @andrepestana-aot @guru-aot suggestions?

Reply:
@ann-aot Found a good read (FYR): https://www.vuemastery.com/blog/vues-watch-vs-watcheffect-which-should-i-use/